### ISSUE TITLE
Open and replace contents of current tab (buffer), like vanilla Vim, configurable

### ISSUE THREAD

**Issue by willhaslett** [VSCodeVim/Vim#4559]
**Describe the solution you'd like**
Provide a setting that overrides Code's behavior of opening files in a new tab, instead replacing the file in the active tab, if it is not dirty.

**Describe alternatives you've considered**
I have a [workaround hack](https://stackoverflow.com/a/60218926/2820927).

**Additional context**
I can't help but express curiosity as to why this feature isn't  a big deal to long-time Vim users. Tabs are fundamentally different in Code, being hierarchically above splits, but the importance of this expected behavior remains the same.

**Comment by J-Fields** [comment]
@willhaslett Does your workaround work, or would there be something to gain (other than convenience) from implementing this in the extension?

**Comment by willhaslett** [comment]
@J-Fields The workaround does work. I was just thinking that this might be worth bringing into the extension because it's the default behavior of vanilla vim, and managing splits/tabs is such a big part of editing, at least for some. Kudos overall, btw. It took this level of Vim emulation (.vimrc!) for me to finally switch to an IDE. :)

**Comment by willhaslett** [comment]
To make things concrete, or perhaps to discover that I'm missing something, here's a screenshot. I would think that working with a layout like this is common, and adding/removing splits with one (or more) tabs in them happens all the time. It's *so* quick in Vim. Adding/removing splits is just as easy with your extension in Code, but the tab management (Code behavior, not the extension) messes everything up.
<img width="1991" alt="Screen Shot 2020-02-21 at 9 36 56 AM" src="https://user-images.githubusercontent.com/2914495/75043959-d2fb5f80-548e-11ea-9b5c-defe1fe61a0b.png">

**Comment by willhaslett** [comment]
Ping. My workaround no longer works for this for some reason. It was a multicommand hack anyway. Since `:edit` is now supported (great!) I'm wondering if my request would now be simple to implement.

Simply want to open a file for editing and have the new file _replace_ the contents of the current tab, if any. Unless there's something funky in my `.vimrc` (for Vim per se) that I'm forgetting, this is default Vim behavior for the `:edit` command and it's central to my workflow.

If someone can point me in the right direction, I'd be happy to write this!

**Comment by J-Fields** [comment]
Yeah the central problem here is that Vim treats tabs as containers for buffers, while VSCode doesn't distinguish between the two. So you can't actually replace the contents of your current tab, you have to close the current one and open a new one in its place.

This is possible (similar to [this](https://github.com/VSCodeVim/Vim/blob/master/src/cmd_line/commands/file.ts#L57)) - I played around with it for just a minute but ran into issues where the editor you're switching from is in preview mode. If that's the case, it gets removed before we try to close it and we end up closing the tab we just opened.

I'll assign myself so I remember to get around to this. If you want to give it a go in the meantime though, feel free - happy to accept PRs.

**Comment by willhaslett** [comment]
I will indeed have a go at it. Offhand, it seems like there are two do-nothing cases; when the previous tab is in preview mode and  when there is no previous tab. I imagine there may be other special cases.
